### DISCUSSION TITLE
Allow push to hidden refs at `pull/*/head`

### DISCUSSION THREAD

**Discussion by JojOatXGME** [community/community#124685] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

I recently learned that you can checkout pull requests locally without adding the fork as another remote.

```shell
> git fetch origin pull/42/head:pull/42
> git checkout pull/42
```

or

```shell
> git config --local --add remote.origin.fetch '+refs/pull/*/head:refs/remotes/origin/pull/*'
> git fetch
> git checkout pull/42
```

However, according to the [documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally?platform=linux#error-failed-to-push-some-refs), these refs are read-only. Assuming the PR is still open and the creator of the PR [allows changes to the pull request branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), it would be nice if I could push to these references.

**Comment by sgleske-ias** [comment]
This is important to me because I'm converting all machine-based GitHub operations to use GitHub app authentication scoped to the organization of private repositories.

A token issued to a GitHub app cannot operate on a private fork of a repository under a user (even though that fork originated from the organization and the user has allowed changes from maintainers).
